financial-templates-lib/ExpiringMultiPartyClient.js

```diff
@@ -1,9 +1,59 @@
+const { delay } = require("./delay");
+
+const ExpiringMultiParty = artifacts.require("ExpiringMultiParty");
+
 // A thick client for getting information about an ExpiringMultiParty.
-export default class ExpiringMultiPartyClient {
-  constructor() {
+class ExpiringMultiPartyClient {
+  constructor(empAddress) {
     this.positions = [];
+    this.sponsorAddresses = [];
+    this.emp = new web3.eth.Contract(ExpiringMultiParty.abi, empAddress);
+    // TODO: Ideally, we'd want to subscribe to events here, but subscriptions don't work with Truffle HDWalletProvider.
```

Comment:
If we did this, would we be able to just get rid of the poll method altogether and only call the update method when new events come in?

Reply:
Maybe. I tried to model the structure here based on dydx's liquidator.

Reply:
Gotcha.